entrypoint.sh

```diff
@@ -51,7 +51,15 @@ SETTINGS_PY="/etc/zulip/settings.py"
 # === initialConfiguration ===
 prepareDirectories() {
     mkdir -p "$DATA_DIR" "$DATA_DIR/backups" "$DATA_DIR/certs" "$DATA_DIR/letsencrypt" "$DATA_DIR/uploads"
-    [ -e /etc/letsencrypt ] || ln -ns "$DATA_DIR/letsencrypt" /etc/letsencrypt
+
+	# See commentary in the Dockerfile about this process.
+	if [ -e /etc/letsencrypt ]; then
+		echo "Found unexpected /etc/letsencrypt in the Docker image, are you using the latest build?" >&2
+		exit 1
+	fi
+	ln -s "${DATA_DIR}/letsencrypt" /etc/letsencrypt
```

Comment:
Add `-n` for clarity. (This would also make the previous check unnecessary, unless we think the verbose error message is important for a case that should be impossible since `entrypoint.sh` is part of the image.)

Reply:
I care a bit less about the verbose message, and more about failing to start the container if we end up in an edge case (that, indeed, *should* be impossible, unless something about the Dockerfile changes in the future without a corresponding change here). `-n` was part of the original file, happy to bring that back but I'm thinking I still want the extra guard in here. (somewhat emulating an `unreachable` declaration from languages that support it...)

Reply:
Oh I guess I meant `ln -Ts`. That would fail if the target already existed.

Reply:
Oh perfect. Just tested that in a throwaway container; does exactly what I wanted. Thanks for introducing me to another new coreutils flag